demo/lego_official_projects/common.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+
+import logging
+import sys
+import time
+from ev3dev.auto import *
+
+log = logging.getLogger(__name__)
+
+
+class Tank(object):
+
+    def __init__(self, left_motor, right_motor):
+
+        for motor in (left_motor, right_motor):
+            if motor not in OUTPUTS:
+                log.error("%s in an invalid output, choices are %s" % (motor, ''.join(OUTPUTS)))
+                sys.exit(1)
+
+        self.left_motor = LargeMotor(left_motor)
+        self.right_motor = LargeMotor(right_motor)
+
+        for x in (self.left_motor, self.right_motor):
+            if not x.connected:
+                log.error("%s is not connected" % x)
```

Comment:
I have this now so that the Device class remembers the kwargs as self.kwargs along with:

    def __str__(self):
        return "%s-%s" % (self.__class__.__name__, self.kwargs.get('address'))

So the output for the code above is now "LargeMotor-outB is not connected"